src/Routing.purs

```diff
@@ -29,11 +39,18 @@ hashes cb =
 
 
 matches :: forall e a. Match a -> (Maybe a -> a -> Eff e Unit) -> Eff e Unit
```

Comment:
This `Eff` feels like it doesn't belong here. You have a few options:

```purescript
matches :: forall e a b. Match a -> (Maybe a -> a -> b) -> Maybe b

matches :: forall e a b. (Monoid b) => Match a -> (Maybe a -> a -> b) -> b
```

With the 2nd option, you'd have to submit a PR for `purescript-monoid` to define `Monoid (Eff eff a)` when `a` is a `Monoid` (which is a sensible monoid to have, just missing).